packages/core/src/bus/EventBus.ts

```diff
@@ -0,0 +1,19 @@
+import { CommandDispatcher } from './CommandDispatcher';
+import { Event } from './Event';
+import { EventDispatcher } from './EventDispatcher';
+import { EventHandler } from './EventHandler';
+
+export interface EventBus extends EventDispatcher, CommandDispatcher {
+  register<T extends Event<R>, R>(
+    type: new (...args: unknown[]) => EventHandler<T>
```

Comment:
Extract a constructor type to `./EventHandler.ts`

```suggestion
    type: EventHandlerConstructor<T, R>
```